### PR TITLE
Add Ecto.DateTime encoder

### PR DESCRIPTION
**Pull request by paulcsmith**
Automatically encode Ecto.DateTime correctly when encoding JSON with Poison

### CODE REVIEW

**Comment by paulcsmith** [comment]
I just saw that there is also `Ecto.Time` and `Ecto.Date`. I will also add those in this PR

File: lib/encoders/date_time.ex [hunk 1]
```diff
@@ -0,0 +1,5 @@
+defimpl Poison.Encoder, for: Ecto.DateTime do
+  def encode(date_time, options) do
+    date_time |> Ecto.DateTime.to_string |> Poison.encode!(options)
```
Comment (josevalim) [inline lib/encoders/date_time.ex]: Thank you! I have actually changed this on Ecto master, we should call `Ecto.DateTime.to_iso8601`. Can you please update the mix.exs to point to latest Ecto? Also, we don't need to call Poison.encode! and we need tests too! :)

Comment (paulcsmith) [inline lib/encoders/date_time.ex]: Ah! I didn't realize there was a new method. I'll add that. Finishing up the tests right now actually :)

Comment (josevalim) [inline lib/encoders/date_time.ex]: Also please rename this to `phoenix_ecto/json.ex`. :) I am going to move the existing code to /html.ex soon.

Comment (josevalim) [inline lib/encoders/date_time.ex]: Ok, I pushed the new code organization if you want to take a look. :)

**Comment by paulcsmith** [comment]
After updating to the latest version of Ecto I get this error when running the `mix test`. Do I need to update something or is Ecto master broken?

```
** (CompileError) test/phoenix_ecto/html_test.exs:44: unknown key :__state__ for struct PhoenixEcto.HTMLTest.User
    (elixir) src/elixir_map.erl:182: :elixir_map."-assert_struct_keys/5-lc$^0/1-0-"/5
    (elixir) src/elixir_map.erl:59: :elixir_map.translate_struct/4
    (stdlib) lists.erl:1352: :lists.mapfoldl/3
```

**Comment by josevalim** [comment]
We no longer have `__state__`. It is `__meta__` now and it has a `state` key in it. :)

**Comment by paulcsmith** [comment]
Thanks :) All tests are passing. This should be ready to go now.

**Comment by josevalim** [comment]
:heart: :green_heart: :blue_heart: :yellow_heart: :purple_heart:

**Pull request merged**